tensorboardX/writer.py

```diff
@@ -344,13 +369,30 @@ def add_image(self, tag, img_tensor, global_step=None):
 
         Args:
             tag (string): Data identifier
-            img_tensor (torch.Tensor): Image data
+            img_tensor (torch.Tensor, numpy.array, or string/blobname): Image data
             global_step (int): Global step value to record
         Shape:
             img_tensor: :math:`(3, H, W)`. Use ``torchvision.utils.make_grid()`` to prepare it is a good idea.
         """
+        if self._check_caffe2(img_tensor):
+            img_tensor = workspace.FetchBlob(img_tensor)
         self.file_writer.add_summary(image(tag, img_tensor), global_step)
 
+    def add_image_with_boxes(self, tag, img_tensor, box_tensor, global_step=None, **kwargs):
+        """Add image boxes data to summary (for Detectron models).
```

Comment:
Is this function specific to detectron?

Reply:
It is used mainly for Detectron, I'm sure that the ability to draw boxes on images would also be useful in other computer vision applications.

Reply:
Then `(for Detectron models)` can be removed

Reply:
Have updated comment to reflect that this functionality could be used for other CV models as well.